### PR TITLE
feat: impl donate func

### PR DESCRIPTION
**Pull request by unicrypto11**
implemented donate sol and token to competition.

### CODE REVIEW

File: programs/soonak_memes_program/src/lib.rs [hunk 1]
```diff
@@ -77,13 +102,19 @@ pub struct StartComp<'info> {
...
 
+
 #[derive(Accounts)]
 pub struct Donate2CompSol<'info> {
     #[account(mut)]
     pub comp: Account<'info, Comp>,
 
     #[account(mut)]
     pub user: Signer<'info>,
+    /// CHECK: safe
```
Comment (hcrypto7) [inline programs/soonak_memes_program/src/lib.rs:113]: write clear safe check report

File: programs/soonak_memes_program/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 use anchor_lang::prelude::*;
-use anchor_spl::token_interface::{TokenAccount, Mint, TokenInterface};
+use anchor_spl::token::{TokenAccount, Token};
```
Comment (hcrypto7) [inline programs/soonak_memes_program/src/lib.rs:2]: use tokeninferface

Comment (hcrypto7) [reply programs/soonak_memes_program/src/lib.rs:2]: Okay. I will use.

File: programs/soonak_memes_program/src/lib.rs (hunk 1 above)
Comment (hcrypto7) [reply programs/soonak_memes_program/src/lib.rs:113]: THank you.

File: programs/soonak_memes_program/src/lib.rs (hunk 2 above)
Comment (unicrypto11) [reply programs/soonak_memes_program/src/lib.rs:2]: thank you for your review and fixing.

**Pull request merged**